views/SignupView.vue

```diff
@@ -18,8 +19,9 @@ const emit = defineEmits(['signup-event'])
 const schema = z.object({
   email: z.string().email('Invalid email'),
   username: z.string(),
-  password: z.string().min(passwordMinimalLength, 'Must be at least '+passwordMinimalLength+' characters')
+  password: z.string().min(passwordMinimalLength, 'Must be at least '+ passwordMinimalLength + ' characters')
 })
+const usernames: string[] = await getAllUsernamesFromFirebase()
```

Comment:
Doing this is dangerous compared to the former way of doing it, if a user a goes in the signupView and waits a bit of it is possible that a new account is created by user b before the user a creates theirs. Usernames won't be updated and they could technically have the same username

Reply:
The thing is if we make this call from the `validate` function, then we emit requests at each change of a single character of this field. Is the danger you mention worth this drawback ?

Reply:
A solution between the two could be to only redo a request at submit time. If the name is taken by a user before the submit of someone else, it will just wrongly says the username is available and refresh the `usernames` list at submit.